Jenkinsfile

```diff
@@ -0,0 +1,104 @@
+@NonCPS
+def jsonParse(def json) {
+	new groovy.json.JsonSlurperClassic().parseText(json)
+}
+
+def nodeVersion = '4.7.3'
+def tiSDKVersion = '5.3.0.GA'
+def androidAPILevel = '23'
+
+def sdkSetup(sdkVersion) {
+  sh 'npm install titanium'
+  sh './node_modules/.bin/ti login travisci@appcelerator.com travisci'
+  echo "Installing ${sdkVersion}"
+  sh "./node_modules/.bin/ti sdk install -d ${sdkVersion} --no-progress-bars"
+  def sdkListJSON = jsonParse(sh(returnStdout: true, script: './node_modules/.bin/ti sdk list -o json'))
+  def titaniumRoot = sdkListJSON['defaultInstallLocation']
+  def activeSDKVersion = sdkListJSON['activeSDK']
+  return sdkListJSON['installed'][activeSDKVersion]
+}
+
+def buildAndroid() {
+  return {
+    node('android-sdk && android-ndk && (osx || linux)') {
+      unstash 'sources'
+      nodejs(nodeJSInstallationName: "node ${nodeVersion}") {
+        def activeSDKPath = sdkSetup(tiSDKVersion)
+
+        // We have to hack to make sure we pick up correct ANDROID_SDK/NDK values from the node that's currently running this section of the build.
+        withEnv(['ANDROID_SDK=', 'ANDROID_NDK=']) {
+          def androidSDK = sh(returnStdout: true, script: 'printenv ANDROID_SDK')
+          def androidNDK = sh(returnStdout: true, script: 'printenv ANDROID_NDK')
+          sh "./node_modules/.bin/ti config android.sdkPath ${androidSDK}"
+          sh "./node_modules/.bin/ti config android.ndkPath ${androidNDK}"
+          dir('android') {
+            writeFile file: 'build.properties', text: """
+titanium.platform=${activeSDKPath}/android
+android.platform=${androidSDK}/platforms/android-${androidAPILevel}
+google.apis=${androidSDK}/add-ons/addon-google_apis-google-${androidAPILevel}
+"""
+            // TODO Use 'appc ti build --build-only'!
+            // if lib folder doesn't exist, create it
+            sh 'mkdir -p lib'
+            // if build folder doesn't exist, create it
+            sh 'mkdir -p build'
+            // if build/docs folder doesn't exist, create it
+            sh 'mkdir -p build/docs'
+            def antHome = tool(name: 'Ant 1.9.2', type: 'ant')
+            withEnv(["PATH+ANT=${antHome}/bin"]) {
+              sh 'ant clean'
+              sh 'ant'
```

Comment:
We should use `ti build -p android --build-only` or the Appc-CLI `appc run -p android --build-only`.